### DISCUSSION TITLE
I edit commands on the terminal too laggy

### DISCUSSION THREAD

**Discussion by xuxz-123** [eclipse-theia/theia#13463] [Q&A]
I used K8S to deploy a theia container. When I opened the theia terminal on my page to execute the command, I found that my input command was too laggy, and my editing file was not. Is there any setting or method to solve it

**Comment by JonasHelming** [comment]
Hi,

what kind of command are you triggering there? Can you provide more details, please? Also, can you reproduce this locally, without running on K8S? In general, if you trigger something in the terminal, this is actually executed in the underlying container, Theia just shows the output. So depending on what kind of command you trigger, this has potentially nothing to do with Theia, but with your container, e.g. the available resources. Did you try triggering the same command in the container outside of Theia?

Best regards,
Jonas

**Reply by xuxz-123** [reply]
Does Theia have high network requirements? Why is it slow for me to open a project directory with only two files

**Reply by msujew** [reply]
Theia deals fairly well with low-bandwidth and/or high latency connections. Most of the data transmission happens via websockets. You can inspect the websocket connection using your browser tools to see what goes wrong.

This is likely an infrastructure issue and nothing we can fix/improve on Theia's side. Does this only happen in a cloud deployment environment or do you have the same issue in a local deployment? How much resources do you assign to each container/pod?

**Comment by xuxz-123** [comment]
Run in the pod of k8s, and the resource is 1c4g，
Input characters in the terminal, and after entering a few consecutive characters, there will be a brief pause

**Reply by msujew** [reply]
Can you inspect the pod via something like `htop` for its resource utilization? Especially while typing in the console. Your symptoms strongly suggest resource starvation inside of the pod.

**Reply by xuxz-123** [reply]
This error is reported during git stage changes
2024-03-18T08:36:25.919Z root ERROR Error: No reply handler for reply with id: 133

**Reply by msujew** [reply]
Ok? But that seems to be unrelated. How about the resource utilization?
